src/components/hoc/withSession.tsx

```diff
@@ -0,0 +1,27 @@
+'use client'
```

Comment:
suggestion: Following DDD, this file would be better placed at `app/domains/auth/hooks.tsx` because well... it's related to authentication. WDYT?

Reply:
sounds good!